pkg/ircslack/channel.go

```diff
@@ -101,7 +101,7 @@ func (c *Channel) IsPublicChannel() bool {
 
 // IsPrivateChannel returns true if the channel is private.
 func (c *Channel) IsPrivateChannel() bool {
-	return c.IsGroup && c.IsPrivate
+	return (c.IsGroup || c.IsChannel) && c.IsPrivate
```

Comment:
I'm often missing messages from others (reactions come in fine) in these such private channels (that it had trouble joining before this change), so I wonder if something else needs to change here.